File: server/src/Models/DMToolkitModels/Items/Bases/ItemDefinitionBase.cs
namespace DMToolkit.API.Models.DMToolkitModels.Items.Bases;

public abstract class ItemDefinitionBase : WorthBase
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public int Weight { get; set; } = 0;
    public string ItemType { get; set; } = "Item";
}

File: flutter-client/dm_toolkit/lib/models/dm_toolkit/items/bases/item_definition_base.dart
import 'package:dm_toolkit/models/dm_toolkit/items/bases/worth.dart';

abstract class ItemDefinitionBase {
  final String id;
  final String name;
  final String description;
  final int weight;
  final String itemType;
  final Worth worth;

  ItemDefinitionBase({
    required this.id,
    required this.name,
    required this.description,
    required this.weight,
    required this.itemType,
    required this.worth
  });
}